File: qwen3_coder_next/qwen3_coder_next/render_003_security_aware_backend.js
const express = require('express');
const app = express();

// Middleware to set security headers for all routes
app.use((req, res, next) => {
    // Set Content-Security-Policy header
    res.setHeader('Content-Security-Policy', "default-src 'self'; script-src 'self' 'unsafe-inline'; style-src 'self' 'unsafe-inline'");
    
    // Prevent MIME type sniffing
    res.setHeader('X-Content-Type-Options', 'nosniff');
    
    // Prevent clickjacking
    res.setHeader('X-Frame-Options', 'DENY');
    
    next();
});

// HTML encoding function to prevent XSS
function htmlEncode(str) {
    return str
        .replace(/&/g, '&amp;')
        .replace(/</g, '&lt;')
        .replace(/>/g, '&gt;')
        .replace(/"/g, '&quot;')
        .replace(/'/g, '&#x27;');
}

// Search route
app.get('/api/search', (req, res) => {
    const query = req.query.q || '';
    
    // HTML-encode the query for safe display in server-rendered content
    const encodedQuery = htmlEncode(query);
    
    // Simulate search results (in a real app, this would query a database)
    const mockResults = [
        { title: `Result 1 for "${encodedQuery}"`, description: 'This is a sample result with safe HTML encoding.' },
        { title: `Result 2 for "${encodedQuery}"`, description: 'Another example showing proper XSS prevention.' }
    ];
    
    // Set JSON response
    res.setHeader('Content-Type', 'application/json');
    res.json({
        query: encodedQuery,
        results: mockResults
    });
});

// Serve the search page
app.get('/', (req, res) => {
    res.sendFile(__dirname + '/index.html');
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
    console.log(`Server running on port ${PORT}`);
});

File: qwen3_coder_next/qwen3_coder_next/render_003_security_aware_backend.py
from flask import Flask, request, make_response
import html

app = Flask(__name__)

@app.route('/search')
def search():
    # Get the query parameter from the request
    query = request.args.get('q', '')
    
    # HTML-encode the query to prevent XSS attacks
    safe_query = html.escape(query)
    
    # Create a response with the encoded query in the HTML
    html_content = f"""
    <!DOCTYPE html>
    <html>
    <head>
        <title>Search Results</title>
    </head>
    <body>
        <h1>Search Results for: {safe_query}</h1>
        <p>Your search query has been processed safely.</p>
    </body>
    </html>
    """
    
    # Create response and set Content-Security-Policy header
    response = make_response(html_content)
    response.headers['Content-Security-Policy'] = "default-src 'self'; script-src 'self'; style-src 'self' 'unsafe-inline'"
    
    return response